frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useEffect } from 'react'
+
+export default function useAutoLogout(timeout: number, onTimeOut: () => void) {
+  const logout = useCallback(() => {
+    clearTimeout()
+    onTimeOut && onTimeOut()
+  }, [onTimeOut])
+
+  useEffect(() => {
+    let logoutTimeout: number
+
+    const setTimeouts = () => {
+      logoutTimeout = setTimeout(logout, timeout)
+    }
+
+    const clearTimeouts = () => {
+      if (logoutTimeout) clearTimeout(logoutTimeout)
+    }
+    const events = [
+      'load',
+      'mousemove',
+      'mousedown',
+      'click',
+      'scroll',
+      'keypress',
+      'touch'
+    ]
+
+    const resetTimeout = () => {
+      clearTimeouts()
+      setTimeouts()
+    }
+
+    const logoutOnResume = () => {
+      clearTimeouts()
+      logout()
+    }
+
+    for (const event of events) {
+      window.addEventListener(event, resetTimeout, { passive: true })
+      window.addEventListener('blur', logoutOnResume, { passive: true })
```

Comment:
should this be outside for loop? now blur handler gets added 7 times

Reply:
Good catch, fixed!